test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.WebEncoders.Testing;
+using Moq;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Localization.Test
+{
+    public class HtmlLocalizerTest
+    {
+        [Fact]
+        public void HtmlLocalizer_UseIndexer_ReturnsLocalizedString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("Hello", "Bonjour");
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s["Hello"]).Returns(localizedString);
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer.Object, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_UseIndexerWithArguments_ReturnsLocalizedString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("Hello", "Bonjour test");
+
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s["Hello", "test"]).Returns(localizedString);
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer.Object, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello", "test"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        public static IEnumerable<object[]> HtmlData
+        {
+            get
+            {
+                yield return new object[] { "Bonjour {0} {{{{ }}", new object[] { "test" }, "Bonjour HtmlEncode[[test]] {{ }" };
+                yield return new object[] { "Bonjour {{0}}", new object[] { "{0}" }, "Bonjour {0}" };
+                yield return new object[] { "Bonjour {0:x}", new object[] { 10 }, "Bonjour HtmlEncode[[a]]" };
+                yield return new object[] { "Bonjour {0:x}}}", new object[] { 10 }, "Bonjour HtmlEncode[[x}]]" };
+                yield return new object[] { "Bonjour {{0:x}}", new object[] { 10 }, "Bonjour {0:x}" };
+                yield return new object[] { "{{ Bonjour {{{0:x}}}", new object[] { 10 }, "{ Bonjour {HtmlEncode[[x}]]" };
+                yield return new object[] { "}} Bonjour {{{0:x}}}", new object[] { 10 }, "} Bonjour {HtmlEncode[[x}]]" };
+                yield return new object[] { "}} Bonjour", new object[] { }, "} Bonjour" };
+                yield return new object[] { "{{ {0} }}", new object[] { 10 }, "{ HtmlEncode[[10]] }" };
+                yield return new object[] { "{0:{{000}}}", new object[] { 10 }, "HtmlEncode[[{010}]]" };
```

Comment:
Mono apparently doesn't deal well with custom format strings, even valid ones. The `HtmlLocalizer_HtmlWithArguments_ReturnsLocalizedHtml(format: "{0:{{000}}}", ...)` case is failing in Travis.

Conditionalize this line i.e. only `yield return` this entry when `!TestPlatformHelper.IsMono`. Will need to add a `using` for the `Microsoft.AspNet.Testing` namespace but this project already references the containing package.